gnovm/pkg/gnolang/realm.go

```diff
@@ -871,6 +907,175 @@ func (rlm *Realm) clearMarks() {
 	rlm.escaped = nil
 }
 
+// assertNoPrivateDeps checks for private dependencies.
+// this take a visited map to avoid infinite recursion on circular references & optimize repeated checks.
+func (rlm *Realm) assertNoPrivateDeps(obj Object, store Store, visited map[Object]struct{}) {
```

Comment:
question: Should we not do the check of private deps on the getChildObject since it does same thing ?